src/pages/playground/Playground.tsx

```diff
@@ -175,13 +197,61 @@ const Playground: React.FC<PlaygroundProps> = props => {
     }
   };
 
+  const uploadLogs = React.useCallback(() => {
+    const assessmentIDs = getLoggedAssessmentIds();
+    const assessmentLogs = getAssessmentLogs();
+    const playgroundLogs = getPlaygroundLogs();
+
+    if (assessmentLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(
+        assessmentIDs.assessmentId,
+        assessmentIDs.questionId,
+        assessmentLogs
+      );
+    }
+
+    if (playgroundLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(playgroundQuestionId, playgroundQuestionId, playgroundLogs);
+    }
+
+    resetPlaygroundLogging(props.sourceChapter, props.externalLibraryName, props.editorValue);
+  }, [props]);
```

Comment:
What happens when the logs are uploaded when the user is still on the playground?

Reply:
localStorage is reset wrt to playground logs **if and only if** the logs are uploaded successfully

Reply:
Which piece of code says so? How do you know if the logs are successfully uploaded?